src/app/_pages/join/join.page.ts

```diff
@@ -1,21 +1,67 @@
 import { Component } from '@angular/core';
 import { ApiService } from '_services/api.service';
+import { Group } from '_models/group.model';
 
 @Component({
   selector: 'join-page',
   templateUrl: 'join.page.html',
   styleUrls: ['join.page.scss'],
 })
 export class JoinPage {
-  public housingGroups: string[] = ['Pittsburgh', 'Oakland', 'Shadyside', 'Squirrel Hill', 'Lawrenceville', 'Southside', 'East Liberty', 'Strip District', 'Mount Washington', 'Downtown', 'Duquesne Heights'];
+  public joinGroups: Group[] = [];
+  public selectedOptions: Group[] = [];
+  public userAccount: any;
+  public currentGroups: any = [];
+
   constructor(
     private apiService: ApiService,
   ) {
-
+    this.getAccount();
+    this.getJoinGroups();
   }
 
   ngOnInit() {
+    this.getJoinGroups();
```

Comment:
getAccount race condition with getJoinGroups. getJoinGroups called twice. Any api calls should happen in ngOnInit. Discussed more in the conversation section.